pkg/oci/util/util.go

```diff
@@ -28,11 +29,11 @@ const (
 )
 
 // MapProviderIDToInstanceID parses the provider id and returns the instance ocid.
-func MapProviderIDToInstanceID(providerID string) string {
+func MapProviderIDToInstanceID(providerID string) (string, error) {
 	if strings.HasPrefix(providerID, providerPrefix) {
-		return strings.TrimPrefix(providerID, providerPrefix)
+		return strings.TrimPrefix(providerID, providerPrefix), nil
 	}
-	return providerID
+	return providerID, errors.Errorf("(%q) is not a valid provider ID", providerID)
```

Comment:
This would be a breaking change for existing deployments since it's optional to have the prefix which is how other clouds work as well. You can't actually run multiple CCMs in a single cluster across clouds. 

 I think a better check would be if empty string return error.

Reply:
Also, once the provider id is set you cannot change it so every node without the prefix would need to be deleted and register again w/ the cluster to fix it.

Reply:
Oh, wasn't aware of that. Thanks. Will change